Apps/GatewayApi/src/Models/UserProfileModel.cs

```diff
@@ -123,6 +123,7 @@ public static UserProfileModel CreateFromDbModel(UserProfile model)
                 IsSMSNumberVerified = !string.IsNullOrEmpty(model.SMSNumber),
                 LastLoginDateTime = model.LastLoginDateTime,
                 ClosedDateTime = model.ClosedDateTime,
+                AcceptedTermsOfService = model.TermsOfServiceId != Guid.Empty,
```

Comment:
This initialization doesn't seem write to me as having a TermsOfServiceId doesn't imply they have accepted the terms of service.  I think any logic like this should be in service code and simple a default boolean initialization of false should be sufficient.  Do we need this for some reason?

Reply:
Having a non-empty `TermsOfServiceId` does imply they have accepted the terms of service.  That is how we are using that database field.

Setting the value here instead of in the service ensures that the `AcceptedTermsOfService` value is set properly for all actions that return a `UserProfileModel` instead of just the `GetUserProfile` action.

Reply:
This code was moved over to the UserProfileModel because both 'UpdateAcceptedTerms' and 'GetUserProfile' needed it.  

Scenario:

1. User logs in and is required to accept terms of service
2. User accepts term of service via 'UpdateAcceptedTerms' - but UpdateAcceptedTerms call does not update boolean AcceptedTermsOfService.  It is defaulted to false.
3. At this point user cannot navigate to the home page because the route requires user to be registered i.e., accepted terms of service.

Question:

If a new Terms of Service is added to the Legal Agreement table, can we not null out the TermsOfServiceId column in User Profile.  Also right now, it looks UserProfile.TermsOfServiceId is a NOT NULL column.